src/app/content/components/Page/contentLinkHandler.spec.ts

```diff
@@ -1,116 +1,183 @@
-import { HTMLAnchorElement } from '@openstax/types/lib.dom';
+import { HTMLAnchorElement, MouseEvent } from '@openstax/types/lib.dom';
 import defer from 'lodash/fp/defer';
 import { book as archiveBook, page } from '../../../../test/mocks/archiveLoader';
 import { mockCmsBook } from '../../../../test/mocks/osWebLoader';
+import { resetModules } from '../../../../test/utils';
 import { assertDocument } from '../../../utils';
 import * as routes from '../../routes';
 import { formatBookData } from '../../utils';
-import { contentLinkHandler, ContentLinkProp } from './contentLinkHandler';
+import { ContentLinkProp } from './contentLinkHandler';
 
 const book = {
   ...formatBookData(archiveBook, mockCmsBook),
   id: 'longidin-vali-dfor-mat1-111111111111',
 };
 
 describe('contentLinkHandler', () => {
-  let handler: ReturnType<typeof contentLinkHandler>;
+  let handler: (e: MouseEvent) => Promise<void>;
   let prop: ContentLinkProp;
   let anchor: HTMLAnchorElement;
 
   beforeEach(() => {
+    resetModules();
     anchor = assertDocument().createElement('a');
 
     prop = {
       book,
       currentPath: '/asdf',
+      hasUnsavedHighlight: false,
       locationState: {} as any,
       navigate: jest.fn(),
       page,
       references: [],
     };
+  });
 
-    handler = contentLinkHandler(anchor, () => prop);
+  describe('without unsaved highlight', () => {
```

Comment:
Those two tests here fail but say that there is no visual difference between expected and what was the function called with. I'm not exactly sure why does that make it fail.

I was able write the test for `.removeAllRanges()` but I'm struggling a lot with mocking the returned value from `showConfirmation` in tests for highlight manager so that there is a test for both confirm/deny scenarios.